### PR TITLE
Added search icon in the top app bar

### PR DESCRIPTION
**Pull request by kamdaou-amalitech**
Added search icon
Search text field is opened when we click on the Search Icon

https://github.com/AmaliTech-Training-Rw/Meeting-Room-booking-App/assets/132997531/39b2eb26-5de0-42ef-8239-ce9f9ddc0241

### CODE REVIEW

File: app/src/main/java/com/amalitech/bookmeetingroom/MainActivity.kt [hunk 1]
```diff
@@ -20,7 +28,22 @@ class MainActivity : ComponentActivity() {
...
+                    topBar = {
+                        BookMeetingTopAppBar(
+                            appState = appState,
+                            title = "Home",
+                            query,
+                            { query = it }
+                        ) { }
+                    }
+                ) {
+                    Text(text = "Test", modifier = Modifier.padding(it))
```
Comment (esthcarelle) [inline app/src/main/java/com/amalitech/bookmeetingroom/MainActivity.kt:45]: @kamdaou-amalitech I created the DeugScreen for you guys to be able to test in the MainActivity

File: app/src/main/java/com/amalitech/bookmeetingroom/MainActivity.kt [hunk 2]
```diff
@@ -20,7 +28,22 @@ class MainActivity : ComponentActivity() {
...
             BookMeetingRoomTheme {
                 val showOnboarding by viewModel.showOnBoarding.collectAsStateWithLifecycle()
-                AppScaffold(shouldShowOnboarding = showOnboarding)
+                val appState = rememberBookMeetingRoomAppState()
+                var query by rememberSaveable {
+                    mutableStateOf("")
+                }
+                Scaffold(
+                    topBar = {
+                        BookMeetingTopAppBar(
```
Comment (esthcarelle) [inline app/src/main/java/com/amalitech/bookmeetingroom/MainActivity.kt:37]: You can add a top app bar button in the debug screen

File: core_ui/src/main/java/com/amalitech/core_ui/components/BookMeetingTopAppBar.kt [hunk 3]
```diff
@@ -65,6 +80,57 @@ fun BookMeetingTopAppBar(
             }
         },
         actions = {
+//            if (
+//                appState.currentDestination?.route == NavigationItem.Users.route
+//                || appState.currentDestination?.route == NavigationItem.BookingRequests.route
+//                || appState.currentDestination?.route == NavigationItem.BookingHistory.route
```
Comment (esthcarelle) [inline core_ui/src/main/java/com/amalitech/core_ui/components/BookMeetingTopAppBar.kt:86]: I guess this need to be removed

File: core_ui/src/main/java/com/amalitech/core_ui/components/BookMeetingTopAppBar.kt [hunk 4]
```diff
@@ -65,6 +80,57 @@ fun BookMeetingTopAppBar(
...
+//            ) {
+            if (!isSearching) {
+                IconButton(onClick = { isSearching = true }) {
```
Comment (esthcarelle) [inline core_ui/src/main/java/com/amalitech/core_ui/components/BookMeetingTopAppBar.kt:89]: Can we have this extracted in its own composable?

File: core_ui/src/main/java/com/amalitech/core_ui/components/BookMeetingTopAppBar.kt [hunk 5]
```diff
@@ -65,6 +80,57 @@ fun BookMeetingTopAppBar(
...
+                        contentDescription = stringResource(
+                            R.string.open_search
+                        )
+                    )
+                }
+            } else {
+                TextField(
+                    value = searchQuery,
+                    onValueChange = {
+                        onSearchQueryChange(it)
```
Comment (esthcarelle) [inline core_ui/src/main/java/com/amalitech/core_ui/components/BookMeetingTopAppBar.kt:101]: Can you rename the it and make it query?

**Review by esthcarelle — CHANGES_REQUESTED** [review]

**Review by ma-za-kpe — COMMENTED** [review]
The search feature only appears in the user composable, according to the figma designs.

File: core_ui/src/main/java/com/amalitech/core_ui/ui/BookMeetingRoomApp.kt [hunk 6]
```diff
@@ -26,7 +26,7 @@ fun BookMeetingRoomApp(
             // Show the top app bar on top level destinations.
             val destination = appState.currentTopLevelDestination
             if (destination != null) {
-                BookMeetingTopAppBar(appState, title)
+                BookMeetingTopAppBar(appState, title, "search", {}) {}
```
Comment (bensek) [inline core_ui/src/main/java/com/amalitech/core_ui/ui/BookMeetingRoomApp.kt:29]: Look into adding the toolbar actions dynamically depending on the screen you are on.

**Review by bensek — CHANGES_REQUESTED** [review]

File: core_ui/src/main/java/com/amalitech/core_ui/components/BookMeetingTopAppBar.kt [hunk 7]
```diff
@@ -31,9 +34,15 @@ import kotlinx.coroutines.launch
...
     appState: BookMeetingRoomAppState,
-    title: String
+    title: String,
+    searchQuery: String? = null,
+    onSearchQueryChange: ((query: String) -> Unit)? = null,
+    onSearchClick: (() -> Unit)? = null
 ) {
     val coroutineScope: CoroutineScope = rememberCoroutineScope()
+    var isSearchTextFieldVisible by rememberSaveable {
+        mutableStateOf(false)
```
Comment (esthcarelle) [inline core_ui/src/main/java/com/amalitech/core_ui/components/BookMeetingTopAppBar.kt:44]: could you please pass this isSearchTextField as a parameter please?

**Review by esthcarelle — CHANGES_REQUESTED** [review]

**Review by esthcarelle — APPROVED** [review]

File: core_ui/src/main/java/com/amalitech/core_ui/components/BookMeetingTopAppBar.kt [hunk 8]
```diff
@@ -80,57 +69,13 @@ fun BookMeetingTopAppBar(
...
-                        unfocusedTextColor = MaterialTheme.colorScheme.onBackground,
-                        focusedContainerColor = MaterialTheme.colorScheme.background,
-                    ),
-                    placeholder = {
-                        Text(stringResource(R.string.search))
-                    }
-                )
-            }
-//            }
+            SearchIcon(
```
Comment (bensek) [inline core_ui/src/main/java/com/amalitech/core_ui/components/BookMeetingTopAppBar.kt:72]: This will be shown on all toolbars which we don't want.

Comment (kamdaou-amalitech) [reply core_ui/src/main/java/com/amalitech/core_ui/components/BookMeetingTopAppBar.kt:72]: In the navigation graph, we can add these 3 variables: searchQuery, onSearchQueryChange, and onClick to specify that the screen should have a topAppBar with a search icon. Please refer to `app/src/main/java/com/amalitech/bookmeetingroom/navigation/DebugNavGraph.kt` to see how I used it. These 3 variables should be provided by the ViewModel in charge of the called screen composable

Comment (kamdaou-amalitech) [reply core_ui/src/main/java/com/amalitech/core_ui/components/BookMeetingTopAppBar.kt:72]: Is it still not dynamic enough?

**Review by bensek — APPROVED** [review]

**Comment by esthcarelle** [comment]
conflicts

**Comment by kamdaou-amalitech** [comment]
@esthcarelle resolved

**Pull request merged**